app/core/projection_engine_job.py

```
@@ -0,0 +1,64 @@
+    registry = get_registry()
+    interval = settings.projection_engine_interval_seconds
+    batch_size = settings.projection_engine_batch_size
+
+    while True:
+        try:
+            async with AsyncSessionLocal() as db:
+                async with db.begin():
+                    projection_repo = ProjectionRepository(db)
+                    event_repo = EventRepository(db)
+
+                    def subject_repo_factory(tenant_id: str):
```

Ruff (0.15.5)

[warning] Missing return type annotation for private function `subject_repo_factory`

(ANN202)